src/functions-and-arrays.js

```diff
@@ -1,38 +1,68 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
-
+function maxOfTwoNumbers(number1, number2) {
+  if (number1 > number2) {
+    return number1;
+  } else {
+    return number2;
+  }
+}
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
-
-
+function findLongestWord(wordarray) {
+  if (wordarray.length === 0) {
+    return null;
+  }
+
+  let longestword = '';
+  for (let i = 0; i < wordarray.length; i++) {
+    if (wordarray[i].length > longestword.length) {
+      longestword = wordarray[i];
+    }
+  }
+  return longestword;
+}
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
-
+function sumNumbers(numbersarray) {
+  let result = 0;
+  for (let i = 0; i < numbersarray.length; i++) {
+    result = result + numbersarray[i];
+  }
+  return result;
+}
 
 // Iteration #3.1 Bonus:
 function sum() {}
 
-
-
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
-
+function averageNumbers(numbersarray) {
+  if (numbersarray.length === 0) {
+    return null;
+  }
+let sum = sumNumbers (numbersarray)
+let numberOfElements= numbersarray.length
+let average = sum / numberOfElements
+return average
+}
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function averageWordLength(wordsarray) {
+  let wordlist = []
+  for (let i = 0; i < wordsarray.length; i++){
+    wordlist.push(wordsarray[i].length)
+  }
+  let average = averageNumbers(wordlist)
+  return average
+}
```

Comment:
This works fine but it creates an extra array. Since the length is just a number we can add all the lengths in a variable instead. But the way you did it also works.

```
function averageWordLength(words) {
  if (words.length === 0) return null; 
  let sum = 0;
  for (let i=0; i < words.length; i++) {
    sum += words[i].length;
  }
  return sum / words.length;
}
```